lib/SecretsCrypto/crypto.c

```diff
@@ -0,0 +1,1543 @@
+/*
```

Comment:
Also, I strongly recommend to run this code under valgrind (or analogues) to check memory leaks at least ones.

Reply:
I ran the tst_secretscrypto unit test under valgrind.  The only leaks I saw were from glib's type registrations, and from glib internals in g_dbus_proxy_new.  That is, none of my manual allocations leaked, as far as I could tell (although I may have missed something, the amount of noise from glib was quite high..)